packages/esm-ward-app/src/ward-workspace/ward-patient-notes/form/notes-form.test.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+import { render, screen } from '@testing-library/react';
+import { createErrorHandler, ResponsiveWrapper, showSnackbar, translateFrom, useSession } from '@openmrs/esm-framework';
+import { savePatientNote } from './notes-form.resource';
+import PatientNotesForm from './notes-form.component';
+import { emrConfigurationMock, mockPatient, mockSession } from '__mocks__';
+import useEmrConfiguration from '../../../hooks/useEmrConfiguration';
+
+const testProps = {
+  patientUuid: mockPatient.uuid,
+  closeWorkspace: jest.fn(),
+  closeWorkspaceWithSavedChanges: jest.fn(),
+  promptBeforeClosing: jest.fn(),
+  setTitle: jest.fn(),
+  onWorkspaceClose: jest.fn(),
+  setOnCloseCallback: jest.fn(),
+};
+
+const mockSavePatientNote = savePatientNote as jest.Mock;
+const mockedShowSnackbar = jest.mocked(showSnackbar);
+const mockedCreateErrorHandler = jest.mocked(createErrorHandler);
+const mockedTranslateFrom = jest.mocked(translateFrom);
+const mockedResponsiveWrapper = jest.mocked(ResponsiveWrapper);
+const mockedUseSession = jest.mocked(useSession);
+
+jest.mock('./notes-form.resource', () => ({
+  savePatientNote: jest.fn(),
+}));
+
+jest.mock('../../../hooks/useEmrConfiguration', () => jest.fn());
+
+const mockedUseEmrConfiguration = jest.mocked(useEmrConfiguration);
+
+mockedUseEmrConfiguration.mockReturnValue({
+  emrConfiguration: emrConfigurationMock,
+  mutateEmrConfiguration: jest.fn(),
+  isLoadingEmrConfiguration: false,
+  errorFetchingEmrConfiguration: null,
+});
+
+test('renders the visit notes form with all the relevant fields and values', () => {
+  renderWardPatientNotesForm();
+
+  expect(screen.getByRole('textbox', { name: /Write your notes/i })).toBeInTheDocument();
+  expect(screen.getByRole('button', { name: /Save/i })).toBeInTheDocument();
+});
+
+test('renders a success snackbar upon successfully recording a visit note', async () => {
+  const successPayload = {
+    encounterProviders: expect.arrayContaining([
+      {
+        encounterRole: emrConfigurationMock.clinicianEncounterRole.uuid,
+        provider: undefined,
+      },
+    ]),
+    encounterType: emrConfigurationMock.visitNoteEncounterType.uuid,
+    location: undefined,
+    obs: expect.arrayContaining([
+      {
+        concept: { display: '', uuid: '162169AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' },
+        value: 'Sample clinical note',
+      },
+    ]),
+    patient: mockPatient.uuid,
+  };
+
+  mockSavePatientNote.mockResolvedValueOnce({ status: 201, body: 'Condition created' });
```

Comment:
```suggestion
  mockSavePatientNote.mockResolvedValue({ status: 201, body: 'Condition created' });
```

In case you might have missed, ignore if intentionally done.